tests/audit/report_test.py

```diff
@@ -0,0 +1,128 @@
+    assert real == expected_real
+    assert false == expected_false
+    delete_all_temporal_files()
+
+
+def count_results(data):
+    real_secrets = 0
+    false_secrets = 0
+    for secret in data:
+        if SecretClassToPrint.from_class(VerifiedResult[secret['category']]) == SecretClassToPrint.REAL_SECRET:  # noqa: E501
+            real_secrets += 1
+        else:
+            false_secrets += 1
+    return real_secrets, false_secrets
+
+
+def baseline_file():
+    # Create our own SecretsCollection manually, so that we have fine-tuned control.
+    url_format = 'http://username:{}@www.example.com/auth'
+    first_secret = 'value1'
+    second_secret = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ'  # noqa: E501
+    random_secret = ''.join(random.choice(string.ascii_letters) for _ in range(8))
+    file_content = 'url = ' + url_format.format(first_secret) \
+        + '\nexample = ' + url_format.format(random_secret) \
+        + '\nlink = ' + url_format.format(first_secret)
+    first_file = create_file_with_content(file_content)
+    file_content = 'url = ' + url_format.format(second_secret) \
+        + '\nexample = ' + url_format.format(random_secret)
+    second_file = create_file_with_content(file_content)
+    secrets = SecretsCollection()
+    secrets[first_file] = {
+        original_potential_secret_factory(
+            type=BasicAuthDetector.secret_type,
+            secret=first_secret,
+            is_secret=True,
+            line_number=1,
+            filename=first_file,
+        ),
+        original_potential_secret_factory(
+            type=BasicAuthDetector.secret_type,
+            secret=random_secret,
+            is_secret=False,
+            line_number=2,
+            filename=first_file,
+        ),
+        original_potential_secret_factory(
+            type=BasicAuthDetector.secret_type,
+            secret=first_secret,
+            is_secret=True,
+            line_number=3,
+            filename=first_file,
+        ),
+    }
+    secrets[second_file] = {
+        original_potential_secret_factory(
+            type=JwtTokenDetector.secret_type,
+            secret=second_secret,
+            is_secret=True,
+            line_number=1,
+            filename=second_file,
+        ),
+        original_potential_secret_factory(
+            type=BasicAuthDetector.secret_type,
+            secret=second_secret,
+            is_secret=False,
+            line_number=1,
+            filename=second_file,
+        ),
+        original_potential_secret_factory(
+            type=BasicAuthDetector.secret_type,
+            secret=random_secret,
+            is_secret=False,
+            line_number=2,
+            filename=second_file,
+        ),
+    }
+
+    f = tempfile.NamedTemporaryFile(delete=False)
+    baseline.save_to_file(secrets, f.name)
+    f.seek(0)
+    CREATED_FILES.append(f.name)
+    return f.name
```

Comment:
You can actually use the built-in context manager to handle this for you. Combined with `pytest.fixture`, this will actually make your code a lot easier to manage.

e.g.

```python
@pytest.fixture
def baseline_filename():
    # create your file contents
    ...

    # then, you can write your files
    with create_file_with_content(...) as first_file, \
            create_file_with_content(...) as second_file, \
            tempfile.NamedTemporaryFile() as baseline_file:
        ...
        baseline.save_to_file(...)
        yield baseline_file.name

# powered by
@contextmanager
def create_file_with_content(content):
    with tempfile.NamedTemporaryFile() as f:
        f.write(content)
        f.seek(0)

        yield f.name

# then
def test_generate_report(..., baseline_filename):
    output = generate_report(baseline_filename, class_to_print)
```

Check out https://github.com/Yelp/detect-secrets/blob/pre-v1-launch/tests/pre_commit_hook_test.py#L53-L55 for an example. This way, you don't have to handle global variables, nor worry about cleaning up after yourself for tests.